build-tools/packages/build-infrastructure/src/test/buildProject.test.ts

```diff
@@ -104,3 +108,87 @@ describe("loadBuildProject", () => {
 		});
 	});
 });
+
+describe("setDependencyRange", () => {
+	const repo = loadBuildProject(testRepoRoot);
+	const main = repo.releaseGroups.get("main" as ReleaseGroupName);
+	assert(main !== undefined);
+	const mainPackages = new Set(main.packages);
+
+	const group2 = repo.releaseGroups.get("group2" as ReleaseGroupName);
+	assert(group2 !== undefined);
+	const group2Packages = new Set(group2.packages);
+
+	const mainWorkspace = repo.workspaces.get("main" as WorkspaceName);
+	assert(mainWorkspace !== undefined);
+	const mainWorkspacePackages = new Set(mainWorkspace.packages);
+
+	const secondWorkspace = repo.workspaces.get("second" as WorkspaceName);
+	assert(secondWorkspace !== undefined);
+	// const secondWorkspacePackages = new Set(secondWorkspace.packages);
+
+	afterEach(async () => {
+		await git.checkout(["HEAD", "--", testRepoRoot]);
+		repo.reload();
+	});
+
+	// it("updates the dependency range to explicit version", async () => {
+	// 	const version = semver.parse("2.0.0");
+	// 	assert(version !== null);
+	// 	await setDependencyRange(mainPackages, group2Packages, version);
+	// 	const allCorrect = main.packages
+	// 		.filter((pkg) => group2Packages.has(pkg))
+	// 		.every((pkg) => pkg.version === "2.0.0");
+	// 	expect(allCorrect).to.be.true;
+	// });
+
+	// it("updates the dependency range given superset workspace package dependencies", async () => {
+	// 	const workspaceVersion = semver.parse(mainWorkspace.rootPackage.version);
+	// 	assert(workspaceVersion !== null);
+	// 	await setDependencyRange(mainPackages, mainWorkspacePackages, workspaceVersion);
+	// 	const allCorrect = main.packages
+	// 		.filter((pkg) => mainWorkspacePackages.has(pkg))
+	// 		.every((pkg) => pkg.version === mainWorkspace.rootPackage.version);
+	// 	expect(allCorrect).to.be.true;
+	// });
+
+	it("updates the dependency range to explicit version", async () => {
+		const version = semver.parse("2.0.0");
+		assert(version !== null);
+		await setDependencyRange(mainPackages, group2Packages, version);
+
+		const allCorrect = main.packages.every((pkg) => {
+			const dependencies = pkg.packageJson.dependencies ?? {};
+
+			const group2PkgDUpdated =
+				"@group2/pkg-d" in dependencies ? dependencies["@group2/pkg-d"] === "2.0.0" : true;
```

Comment:
```suggestion
			const group2PkgDUpdated = dependencies.?["@group2/pkg-d"] === "2.0.0";
```

I think this works.

Reply:
I've updated the check to use nullish coalescing like so `const group2PkgDUpdated = (dependencies["@group2/pkg-d"] ?? "2.0.0") === "2.0.0";`

for these checks, I want the default value to be True since we are scanning all packages, however, not all packages has the changed dependency